2023/incoming/santa-tackles-bitcoin-part-two.pod

```diff
@@ -0,0 +1,126 @@
+Author: Bartosz Jarzyna <contact@bbrtj.eu>
+Title: Santa tackles Bitcoin, part two
+Topic: Bitcoin::Crypto
+
+=encoding utf8
+
+It was a new day in Lapland! Well, sort of... it was still dark outside due to
+polar night, but our elf friend McJingles was well-rested and full of energy!
+Last day he learned all about how to make a transaction using
+L<Bitcoin::Crypto>, but he forgot to prepare the private keys to be gifted by
+Santa. Elves may be very hard-working, but they are also forgetful!
+
+He quickly came up with a plan: the private keys will probably be printed on
+lovely decorated Christmas gift cards. All he had to do is provide printable
+representations of the private keys and the other elves can do all the
+printing. McJingles decided it will for now be enough to print all the keys as
+those Wallet Import Format things he had used before to get access to the
+initial 50 BTC. The standard output of the script can be redirected to a file
+with no additional effort.
+
+=begin perl
+
+use v5.36;
+use Bitcoin::Crypto qw(btc_extprv);
+
+# Ask for mnemonic and password. Keeping them in a file makes them much easier
+# target for malware and hackers!
+say 'Please provide the mnemonic and password, each in separate lines:';
+my $mnemonic = readline STDIN;
+my $password = readline STDIN;
+
+# Remove the newline from readline
+chomp $mnemonic;
+chomp $password;
+
+# recover a key from input
+my $master_key = btc_extprv->from_mnemonic($mnemonic, $password);
+
+# how many presents?
+my $number_of_kids = 346;
+
+# print all the keys in WIF format
```

Comment:
Should we have a brief explanation of "WIF"?

Reply:
It is explained a bit up above, I added abbreviation there in parenthesis so that reader will know WIF = Wallet Import Format and it's used to access coins.

Reply:
Thanks! I think it's helpful to have it here again, as people like me might not remember it from the previous article. 😄